service/controller/resource/alerting/alertmanager/resource.go

```diff
@@ -0,0 +1,200 @@
+package alertmanager
+
+import (
+	"net/url"
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	monitoringv1client "github.com/prometheus-operator/prometheus-operator/pkg/client/versioned"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/controller/resource/generic"
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+var replicas int32 = 1
+
+const (
+	Name = "alertmanager"
+)
+
+type Config struct {
+	Client monitoringv1client.Interface
+	Logger micrologger.Logger
+
+	Address     string
+	CreatePVC   bool
+	LogLevel    string
+	StorageSize string
+	Version     string
+}
+
+func New(config Config) (*generic.Resource, error) {
+	clientFunc := func(namespace string) generic.Interface {
+		c := config.Client.MonitoringV1().Alertmanagers(namespace)
+		return wrappedClient{client: c}
+	}
+
+	c := generic.Config{
+		ClientFunc:    clientFunc,
+		Logger:        config.Logger,
+		Name:          Name,
+		GetObjectMeta: getObjectMeta,
+		GetDesiredObject: func(v interface{}) (metav1.Object, error) {
+			return toAlertmanager(v, config)
+		},
+		HasChangedFunc: hasChanged,
+	}
+	r, err := generic.New(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r, nil
+}
+
+func getObjectMeta(v interface{}) (metav1.ObjectMeta, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return metav1.ObjectMeta{}, microerror.Mask(err)
+	}
+
+	return metav1.ObjectMeta{
+		Name:      key.ClusterID(cluster),
+		Namespace: key.Namespace(cluster),
```

Comment:
Shouldn't this remain in `monitoring` NS? I mean, it's still a "global" resource, and AFAIK `$INSTALLATION-prometheus` is for resources specific to the Prometheus instance dedicated to the MC.

Reply:
I can always move it there of course